source/slang/slang-state-serialize.cpp

```diff
@@ -1072,10 +1074,234 @@ struct LoadContext
+            case SLANG_DEBUG_INFO_LEVEL_MINIMAL:        cmd.addArg("-g1"); break;
+            case SLANG_DEBUG_INFO_LEVEL_MAXIMAL:        cmd.addArg("-g3"); break;
+            default: break;
+        }
+
+        switch (SlangOptimizationLevel(requestState->optimizationLevel))
+        {
+            case SLANG_OPTIMIZATION_LEVEL_NONE:         cmd.addArg("-O0"); break;
+            case SLANG_OPTIMIZATION_LEVEL_DEFAULT:      cmd.addArg("-O");  break;
+            case SLANG_OPTIMIZATION_LEVEL_HIGH:         cmd.addArg("-O2"); break;
+            case SLANG_OPTIMIZATION_LEVEL_MAXIMAL:      cmd.addArg("-O3"); break;
+            default: break;
+        }
+
+        //spSetOutputContainerFormat(externalRequest, SlangContainerFormat(requestState->containerFormat));
+
+        switch (SlangPassThrough(requestState->passThroughMode))
+        {
+            case SLANG_PASS_THROUGH_NONE: break;
+            default:
+            {
+                cmd.addArg("-pass-through");
+                cmd.addArg(getPassThroughName(SlangPassThrough(requestState->passThroughMode)));
+                break;
+            }
+        }
+
+        //request->getBackEndReq()->useUnknownImageFormatAsDefault = requestState->useUnknownImageFormatAsDefault;
+        //request->getBackEndReq()->obfuscateCode = requestState->obfuscateCode;
+        //request->getFrontEndReq()->obfuscateCode = requestState->obfuscateCode;
+
+        switch (requestState->defaultMatrixLayoutMode)
+        {
+            case SLANG_MATRIX_LAYOUT_ROW_MAJOR:     cmd.addArg("-matrix-layout-row-major"); break;
+            case SLANG_MATRIX_LAYOUT_COLUMN_MAJOR:  cmd.addArg("-matrix-layout-column-major"); break;
+            default: break;
+        }
+    }
+
+    // Add the target requests
+    {
+        for (Index i = 0; i < requestState->targetRequests.getCount(); ++i)
+        {
+            TargetRequestState& src = base.asRaw(requestState->targetRequests[i]);
+
+            cmd.addArg("-target");
+            cmd.addArg(getCodeGenTargetName(CodeGenTarget(src.target)));
+
+            cmd.addArg("-profile");
+            cmd.addArg(Profile(src.profile).getName());
+
+            if (src.targetFlags & SLANG_TARGET_FLAG_PARAMETER_BLOCKS_USE_REGISTER_SPACES)
+            {
+                cmd.addArg("-parameter-blocks-use-register-spaces");
+            }
+
+            switch (src.floatingPointMode)
+            {
+                case FloatingPointMode::Fast:
+                {
+                    cmd.addArg("-fp-mode");
+                    cmd.addArg("fast");
+                    break;
+                }
+                case FloatingPointMode::Precise:
+                {
+                    cmd.addArg("-fp-mode");
+                    cmd.addArg("precise");
+                    break;
+                }
+                default: break;
+            }
+
+#if 0
+            // If there is output state (like output filenames) add here
+            if (src.outputStates.getCount())
```

Comment:
This is definitely a hard part of things because the current command-line logic has rules about the ordering of `-o` options that associate them with particular entry points. We'd probably need to interleave the options that specify output files with the options that specify entry points so that the correct per-entry-point files are specified.

In practice, though, I expect the 99% case is that users don't ever use `slangc` to compile multiple entry points, or for multiple targets, in one invocation.

If the goal here is to create dummy `-o` lines for an arbitrary compiler request (possibly one created using the API), then I fear that task is impossible with the current command-line parsing. I don't think there is any way to have both multiple entry points *and* multiple targets with output files right now.